src/Framework/TestCase.php

```diff
@@ -1813,43 +1819,235 @@ private function markSkippedForMissingDependency(ExecutionOrderDependency $depen
         $this->status = TestStatus::skipped($message);
     }
 
+    private function outputBufferingCallback(string $output, int $phase): string
+    {
+        // assign it here to not get output from uncleanable children at the end
+        // as well as to ensure we have all output available to check
+        // if any children buffers have a low chunk size and already returned data
+        // or ob_flush was called
+        if ($this->outputBufferingActive) {
+            $this->output .= $output;
+
+            if (($phase & PHP_OUTPUT_HANDLER_FINAL) === PHP_OUTPUT_HANDLER_FINAL) {
+                // don't handle, since we report an error already if our handler is missing
+                // since it's inconsistent here with ob_end_flush/ob_end_clean
+                return '';
+            }
+
+            if (($phase & PHP_OUTPUT_HANDLER_CLEAN) === PHP_OUTPUT_HANDLER_CLEAN) {
+                $this->outputBufferingFlushed = PHP_OUTPUT_HANDLER_CLEAN;
+            } elseif (($phase & PHP_OUTPUT_HANDLER_FLUSH) === PHP_OUTPUT_HANDLER_FLUSH) {
+                $this->outputBufferingFlushed = PHP_OUTPUT_HANDLER_FLUSH;
+            }
+        }
+
+        return '';
+    }
+
+    // private to ensure it cannot be restarted after ending it from outside this class
     private function startOutputBuffering(): void
     {
-        ob_start();
+        ob_start([$this, 'outputBufferingCallback']);
 
         $this->outputBufferingActive = true;
         $this->outputBufferingLevel  = ob_get_level();
     }
 
+    /**
+     * @throws Exception
+     * @throws NoPreviousThrowableException
+     */
     private function stopOutputBuffering(): bool
     {
-        $bufferingLevel = ob_get_level();
+        $bufferingLevel            = ob_get_level();
```

Comment:
At first, it is a bugfix, so it must target the lowest supported and problematic branch.

Also, it seems this PR fixes several ob issues. I would like to see a separate commit for each issue ("Fix ob is not cleanable", ...) so each can be reviewed individually.

Reply:
@mvorisek While I appreciate the contributions you have made to this project over the years, I do not like how you sometimes interact in discussions.

To somebody like @kkmuffme, whose first pull request this is, it must sound like you are a maintainer for this project when you write statements such as "it must target the lowest supported and problematic branch" or "I would like to see a separate commit for each issue" and that they have to do what you tell them to do. I can only imagine how frustrating it must be for @kkmuffme to read your comments and then do not get answers to the questions they asked in response.

If you look at the [Supported Versions of PHPUnit](https://phpunit.de/supported-versions.html) page then you will see that as of today, PHPUnit 10.5 is the oldest version of PHPUnit that still receives bug fixes. This page is linked from https://github.com/sebastianbergmann/phpunit/blob/main/.github/CONTRIBUTING.md#workflow-for-pull-requests.

As I wrote in https://github.com/sebastianbergmann/phpunit/pull/5852#issuecomment-2422906999, we spent quite some time yesterday trying to understand the problems @kkmuffme intends to solve as well as their proposed changes to solve them. The way I understand the proposed changes, they do not make much sense in isolation from each other. Therefore, I do not require separate commits to review.

I hope that, as outlined in https://github.com/sebastianbergmann/phpunit/pull/5852#issuecomment-2422906999, a simpler solution than the one proposed by @kkmuffme is possible. However, if the problems in questions can only be solved using the changes proposed here then I will gladly merge this pull request. However, given the complexity of the proposed changes I am reluctant to accept this changes into the `10.5` branch for PHPUnit 10.5. I would rather include them in the next minor version, which would be PHPUnit 11.5.

Reply:
@sebastianbergmann thank you for your feedback. I have written my post mainly because I worked on this and this PR was/is too complex. It simply felt/feels very long to me - in the https://github.com/sebastianbergmann/phpunit/issues/5851 issue there are multiple problems mentioned, so to help me, you, and all others to understand the changes, I asked/ask to commit each problem with minimal fix&test separately.

Only then I would be willing to spend my time to review each LoC and probably would propose changes to simplify the PR as you did yesterday. Kudos to you.

I hope we can agree on this. Not only because of me, but also because the core code needs to be readable as much as possible to make it easy to understand while debugging any other ob issue.

I am rereading your post again and this is basically what you have written in your last paragraph. That you are unsure about the changes proposed here.

> it must sound like you are a maintainer for this project when you write statements such as "it must target the lowest supported and problematic branch" ... and that they have to do what you tell them to do

Does it matter who expresses the truth about the contribution guidelines? Or is the main issue you had with post not mentioning/linking https://github.com/sebastianbergmann/phpunit/blob/main/.github/CONTRIBUTING.md#workflow-for-pull-requests?